### ISSUE TITLE
[1.10.2] [Feature Suggestions] Snow's Really Quick List

### ISSUE THREAD

**Issue by SnowShock35** [McJty/TheOneProbe#78]
Hello McJty,

Here's your super duper really quick awesome list of additions for TOP.

> "@SnowShock35 If you're fast you can get the feature request in for release. VERY fast :-)" - @McJty

You asked for it. You got it  

- [x] Lever State (On/Off)
- [x] Pet's Owner
- [x] Analog Redstone Signal (Not sure if that's been added yet
- [x] Repeater's Delay & Comparator mode
- [x] Growth percentage on plants should perhaps show in red/yellow until mature and ready to harvest
- [x] Brewing stand blaze powder remaining or brew time remaining
- [x] Colour of Wolf's collar 
- [ ] Mob's Armor Level (Could use the mc shield icon to represent it)
- [ ] Daylight Sensor Mode
- [ ] What entity is riding in a Minecart (If Minecart with chest what's the inventories contents)
- [ ] Farmland should show as farmland and whether hydrated or not
- [ ] Current record in a jukebox
- [ ] Beacon's active ability
- [ ] Cauldrons content
- [ ] Localize Entities (Item Frames, Minecarts etc. Infested Zombies)
- [ ] Monster Eggs should have a config (On by default) to display as their regular block type
- [ ] Show what a mob is holding if anything (E.g. zombie with a shovel)
- [ ] Flower Pot doesn't show in TOP if it has something inside (Vanilla thing?) 
- [ ] Armour Stands really shouldn't show health. Despite being an invulnerable entity it's a tad weird
- [ ] Colour of Sheep's Wool and whether it's there or not (i.e. if the sheep has been sheared)
- [ ] More use of colours (E.G. if something is off/bad make red, if something is on/good make it green and in between yellow.) 
- [ ] Monster Spawner mob type
- [ ] Show whether a hopper is locked or not (Could be helpful from a debugging perspective when doing redstone) I've known people to have occasions where the hopper is locked but it's not entirely obvious where from.
- [ ] End Portal is kinda not localized. . . http://ss.snowshock35.com/2016/11/1478522763-c323442f.png
- [ ] In debug mode (F3+H) TOP should perhaps show the item id (E.g. minecraft:stone) and perhaps ore dictionary list for it (on shift)

**Comment by McJty** [comment]
1. Lever state is already implemented. Doesn't that work for you?
2. That's already implemented

About 2 that's a bit more complicated. The problem is that the owner is stored as an UUID and when the player is not online I have no way (as far as I know) to get the name of that player. So I would only be able to show the owner if the player is actually online which seems a bit weird.

**Comment by SnowShock35** [comment]
Yeah I understand that could be an issue. Would be weird just seeing a UUID and storing the name of the user could just be unnecessary hassle

**Comment by SnowShock35** [comment]
@McJty The last one on the list would be very nice to have 😇 :heart:

**Comment by McJty** [comment]
Repeater and comparator: done

**Comment by SnowShock35** [comment]
Woot :D

**Comment by McJty** [comment]
Wolf collar: done

**Comment by McJty** [comment]
Growth level yellow when not mature. Otherwise shows 'Mature': done

BTW, daylight sensor is already visible through redstone value

**Comment by SnowShock35** [comment]
In regards to the daylight sensor, I was leaning towards whether it could show the difference between daylight mode: http://ss.snowshock35.com/2016/11/1478523425-a9e2d813.png and night mode: http://ss.snowshock35.com/2016/11/1478523474-c2bac5a9.png I mean I imagine you'll not be able to change the icon that appears in TOP because the moonlight sensor is only another state of the daylight sensor and doesn't actually have an item form but if it was possible to see what state it was in then display that in TOP that could be neat :D

**Comment by McJty** [comment]
Brewing stand info: done

**Comment by McJty** [comment]
Ok, need to go to next mod so rest of the features will be for later :-)

**Comment by SnowShock35** [comment]
Okies :D

**Comment by McJty** [comment]
Animal/horse owner: done
